whole-tale/girder_wholetale
# Publish refactor

Counterpart PR to https://github.com/whole-tale/gwvolman/pull/93. This PR:

* Adds `PUT /tale/:id/publish` as a generic publishing endpoint, that only takes a `repository` as an argument
* Allows to configure repositories where Tales can be deposited.
* Provides `GET /repository` per #368 

How to test?
1. Deploy along with https://github.com/whole-tale/gwvolman/pull/93.
1. Create a Tale
1. Using swagger hit `GET /repository` and confirm it returns an empty array
1. Add your `sandbox.zenodo.org` api key via https://girder.local.wholetale.org/#ext_keys
1. Using swagger hit `GET /repository` and confirm it returns `["sandbox.zenodo.org"]`
1. Using swagger publish your Tale with `repository` == "sandbox.zenodo.org"
1. Authenticate with DataONE stage-2 via https://girder.local.wholetale.org/#ext_keys
1. Using swagger hit `GET /repository` and confirm it returns `["https://.../d1/mn", "sandbox.zenodo.org"]`
1. Using swagger publish your Tale with `repository` == "https://dev.nceas.ucsb.edu/knb/d1/mn"
